mathics/builtin/scoping.py

```diff
@@ -629,7 +631,7 @@ class BeginPackage(Builtin):
                  Append[System`Private`$ContextPathStack, $ContextPath];
              $ContextPath = {context, "System`"};
              $Packages = If[MemberQ[System`$Packages,$Context],
-                            None,
+                            $Packages,
```

Comment:
I think I've run into this problem a bit too, so it might be good to get this into master.